snowshu/core/graph_set_runner.py

```diff
@@ -70,10 +70,59 @@ def make_executables(graphs) -> List[GraphExecutable]:
         # the processing does not halt on exceptions
 
         # Tables need to come first to prevent deps deadlocks with views
+
         for graphs in [table_graph_set, view_graph_set]:
             with ThreadPoolExecutor(max_workers=threads) as executor:
-                for executable in make_executables(graphs):
-                    executor.submit(self._traverse_and_execute, executable)
+                def process_executables(executables, retries, wait_time=3):
+                    error = None
+                    re_executables = []
+                    futures = [executor.submit(self._traverse_and_execute, executable)
+                               for executable in executables]
+
+                    time.sleep(wait_time)
+                    wait_list = []
+                    for future, executable in zip(futures, executables):
+                        if future.done():
+                            if exception := future.exception():
+                                # if exception in thread happened - record reference to it and append
+                                # actual executable to a list for re-execution
+                                error = future.result
+                                logger.warning("Concurrent thread finished work with exception:\n%s: %s",
+                                               str(exception.__class__),
+                                               str(exception))
+                                re_executables.append(executable)
+                        else:
+                            wait_list.append((future, executable))
+                    while wait_list:
```

Comment:
this can be reworked not to have duplicate code, but will require additional nested function and will cause complexity raise

Reply:
Would it make sense to only have a nested function for this duplicated code instead of the `process_executables` and `make_executables` functions, and extract `process_executables` into its own separate function (with adjusted inputs)?

That way, we'd only have one level of nesting inside `process_executables` and no nesting in `execute_graph_set`?